support CC=clang

Running buildnb.sh fails when it reaches the stage where binaries are supposed to be built:

```
#   compile  cat/cat.o
/home/pooka/tmp/rumprun-posix/rumptools/bin/x86_64--netbsd-clang -O2 -g -fPIC -std=gnu99 -D__NetBSD__   -std=gnu99 -Werror    -I/home/pooka/tmp/rumprun-posix/rumptools/compat/include -D__NetBSD__ -Ulinux -U__linux -U__linux__ -U__gnu_linux__ --sysroot=/home/pooka/tmp/rumprun-posix/rumptools/dest  -c    /home/pooka/tmp/rumprun-posix/rumpsrc/bin/cat/cat.c
: ctfconvert -g -L VERSION cat.o
#      link  cat/cat.ro
/home/pooka/tmp/rumprun-posix/rumptools/bin/x86_64--netbsd-clang  -nostdlib -r -Wl,-dc -o cat.ro cat.o
clang: warning: argument unused during compilation: '-r'
/usr/bin/ld: warning: cannot find entry symbol _start; defaulting to 00000000004001c0
cat.o: In function `main':
/home/pooka/tmp/rumprun-posix/rumpsrc/bin/cat/cat.c:80: undefined reference to `setprogname'
/home/pooka/tmp/rumprun-posix/rumpsrc/bin/cat/cat.c:81: undefined reference to `__setlocale50'
/home/pooka/tmp/rumprun-posix/rumpsrc/bin/cat/cat.c:110: undefined reference to `__sF'
...
```

--- a/Makefile
+++ b/Makefile
@@ -145,7 +145,7 @@ weakasm.map:	${RUMPSRC}/lib/libc/sys/Makefile.inc ${RUMPMAKE}
 define NBUTIL_templ
 rumpobj/${1}/${2}.ro:
 	( cd ${RUMPSRC}/${1} && ${RUMPMAKE} obj && \
-	    ${RUMPMAKE} LIBCRT0= BUILDRUMP_CFLAGS="-fPIC -std=gnu99 -D__NetBSD__ ${CPPFLAGS.${2}}" ${2}.ro )
+	    ${RUMPMAKE} LDFLAGS=-Wl,-r LIBCRT0= BUILDRUMP_CFLAGS="-fPIC -std=gnu99 -D__NetBSD__ ${CPPFLAGS.${2}}" ${2}.ro )
 
 NBLIBS.${2}:= $(shell cd ${RUMPSRC}/${1} && ${RUMPMAKE} -V '$${LDADD}' | sed 's/-L\S*//g')
 LIBS.${2}=$${NBLIBS.${2}:-l%=rump/lib/lib%.a}

